back/src/Model/RecordingManager.ts

```diff
@@ -0,0 +1,148 @@
+    isRecording: boolean;
+    destroy(): void;
+}
+
+export class RecordingManager implements IRecordingManager {
+    private _isRecording: boolean = false;
+    private _user: SpaceUser | undefined;
+
+    constructor(
+        private readonly communicationManager: ICommunicationManager,
+        private readonly space: ICommunicationSpace
+    ) {}
+
+    public async startRecording(user: SpaceUser, userUuid: string): Promise<void> {
+        if (this._isRecording) {
+            throw new Error("Recording already started");
+        }
+        this._isRecording = true;
+        const currentState = this.communicationManager.currentState;
+        this._user = user;
+
+        if (this.isRecordableState(currentState)) {
+            await this.executeRecording(currentState, user, userUuid);
+        } else {
+            await this.switchToLivekitAndRecord(user, userUuid);
+        }
+
+        this._isRecording = true;
+    }
+
+    private async executeRecording(
+        recordableState: IRecordableState,
+        user: SpaceUser,
+        userUuid: string
+    ): Promise<void> {
+        try {
+            this._isRecording = true;
+            await recordableState.handleStartRecording(user, userUuid);
+        } catch (error) {
+            this._isRecording = false;
+            this._user = undefined;
+            throw error;
+        }
+    }
+
+    private async switchToLivekitAndRecord(user: SpaceUser, userUuid: string): Promise<void> {
+        const livekitState = await createLivekitState(
+            this.space,
+            user.playUri,
+            this.communicationManager.users,
+            this.communicationManager.usersToNotify
+        ); // await is for room creation
+        this.communicationManager.setState(livekitState);
+
+        await this.executeRecording(livekitState, user, userUuid);
+    }
+
+    public async stopRecording(user: SpaceUser): Promise<void> {
+        if (!this._isRecording) {
+            console.warn("No recording is currently active.");
+            return;
+        }
+
+        if (this._user?.spaceUserId !== user.spaceUserId) {
+            throw new Error("User is not the one recording");
+        }
+
+        const currentState = this.communicationManager.currentState;
+
+        if (this.isRecordableState(currentState)) {
+            await currentState.handleStopRecording();
+        }
+
+        this._isRecording = false;
+        this._user = undefined;
+    }
+
+    public handleAddUser(user: SpaceUser): void {
+        if (!this._isRecording) {
+            return;
+        }
+
+        this.space.dispatchPrivateEvent({
+            spaceName: this.space.getSpaceName(),
+            senderUserId: user.spaceUserId,
+            receiverUserId: user.spaceUserId,
+            spaceEvent: {
+                event: {
+                    $case: "startRecordingMessage",
+                    startRecordingMessage: {},
+                },
+            },
+        });
```

Comment:
I'm wondering....

Instead of sending a private message to users connecting, couldn't we instead rely on the Space metadata to tell is a space is being recorded or not? That could be simpler no?